scene/gui/tree.cpp

```diff
@@ -797,7 +814,11 @@ void TreeItem::move_before(TreeItem *p_item) {
 		parent->children_cache.clear();
 	} else {
 		parent->first_child = this;
-		parent->children_cache.insert(0, this);
+		// If the cache is empty, it has not been built but there
```

Comment:
I believe it is correct now.  Why are you asking for changes?

Reply:
And regarding why you had that condition there, I think it makes total sense.  Moving one item doesn't need to invalidate the cache.   Looks like you just didn't consider the case of the cache being uninitialized.   Do you think the code as I modified it still has sketchy boundary conditions?

Reply:
I mean, the optimization is done only for a specific case, in any other case the cache is cleaned. I think that your adjustments are correct, but I'm not sure if the optimization is worth the extra code.

Reply:
Are you intending to approve it though?

I like it this way.  I am building the new Debugger views right now for threaded debugging.  I have column sorting by clicking on headers.  Also, as threads come and go or information changes, things get reshuffled.  Itemst are always moving, which is how I got stuck on these bugs to begin with.   #editor said that there may not be any previous sortable tables in the product, so maybe that's why this took so long to discover.

Reply:
Ok, let's agree on this change, since it properly fixes the bug and that's the important part.

> Are you intending to approve it though?

My approval has near zero weight here, but I'll probably do that anyway.